ident/types.go

```diff
@@ -103,17 +131,30 @@ type Pool interface {
 	// string.
 	StringTag(name, value string) Tag
 
+	// GetTags will create a new array of tags and return it.  When the context
+	// closes the tags array and any tags contained will be finalized.
+	GetTags(ctx context.Context) Tags
```

Comment:
same here, would be useful to have the ability to have an entry point which takes `ctx+tagIter` and gives back a Tags.